enterprise/backend/src/metabase_enterprise/serialization/v2/deserialize.clj

```diff
@@ -0,0 +1,79 @@
+(ns metabase-enterprise.serialization.v2.deserialize
+  (:require [metabase-enterprise.serialization.v2.models :as serdes.models]
+            [metabase.models.serialization.base :as serdes.base]
+            [metabase.models.serialization.hash :as serdes.hash]
+            [toucan.db :as db]
+            [toucan.models :as models]))
+
+;; The deserialization source is a two-arity function: (src) returns a list of all file names, (src path) returns the
+;; contents of that file, converted to EDN.
+;; Therefore an in-memory source can just wrap a {path EDN-contents} map.
+(defn- scan-ids [{:keys [entity_id] :as entity}]
+  (let [pk (get entity (models/primary-key entity))]
+    (cond-> {:by-identity-hash {(serdes.hash/identity-hash entity) pk}}
+      entity_id (assoc :by-entity-id {entity_id pk}))))
+
+(defn- deserialization-prescan-model [model]
+  (transduce (map scan-ids) (partial merge-with merge) {:by-entity-id {} :by-identity-hash {}}
+             (db/select-reducible model)))
+
+(defn- deserialization-prescan []
+  (into {} (for [model serdes.models/exported-models]
+             [(name model) (deserialization-prescan-model model)])))
+
+(defn- path-parts [path]
+  (->> (java.nio.file.Paths/get path (into-array String []))
+       (.iterator)
+       (iterator-seq)
```

Comment:
```suggestion
       .iterator
       iterator-seq
```

If you don't pass any args the `()` is optional and included automatically in the threading macros

Reply:
This code is on ice. Duly noted for future style.